runtime/commonMain/src/kotlinx/serialization/internal/CollectionSerializers.kt

```diff
@@ -140,27 +128,74 @@ sealed class MapLikeSerializer<TKey, TVal, TCollection, TBuilder: MutableMap<TKe
     }
 }
 
+public abstract class PrimitiveArraySerializer<Element, Array, Builder : PrimitiveArraySerializer<Element, Array, Builder>.Builder>
+internal constructor(
+    primitiveSerializer: KSerializer<Element>,
+    primitiveDescriptor: PrimitiveDescriptor
+) : ListLikeSerializer<Element, Array, Builder>(primitiveSerializer) {
+    final override val descriptor = PrimitiveArrayDescriptor(primitiveDescriptor)
+
+    abstract inner class Builder internal constructor() {
```

Comment:
It would not simplify things much, just one generic parameter will be shorter

Reply:
It would be also simplier to discover – since it is a type argument of a public class, it would be impossible to make builders (implementation detail) internal

Reply:
Now we get rid of one more nesting level (that does not help in readability and maintainability).
Making them internal is not an option, because the type is exposed in generic bounds :(

But It would be nice top mark them as `@InternalSerializationApi` when #541 is merged